tracee-ebpf/tracee/tracee.go

```diff
@@ -1108,6 +1151,35 @@ func (t *Tracee) processEvent(ctx *context, args map[argTag]interface{}) error {
 	return nil
 }
 
+func getFileHash(fileName string) string {
+	f, _ := os.Open(fileName)
+	if f != nil {
+		defer f.Close()
+		h := sha256.New()
+		_, _ = io.Copy(h, f)
+		return hex.EncodeToString(h.Sum(nil))
+	}
+	return ""
+}
+
+func (t *Tracee) updateProfile(ctx *context, sourceFilePath string, sourceFileCtime int64) {
+	sourceFileSHA := getFileHash(sourceFilePath)
```

Comment:
can we move the hash calculation to the end of the trace please? to minimize overhead in the critical path

Reply:
Do you mean call `getFileHash` after the pipeline ends but before we save the stats to disk? If so, that would leave a window of time open when the event is captured till the profile is written on disk. During this time the file can be modified and we wouldn't know about the hash prior to modification.

Capturing hash at event capture time mitigates the above.

Reply:
we already capture the binaries at the time of execution (-c exec). they could be used for calculating the hash right?

Reply:
I see yeah that would work as well. Taking the hash of captured artifacts. Although there is a possibility that something malicious removes those captured artifacts as soon as they are created (as the location is known) but it's a sidetrack. 

I'll implement taking the hash of captured objects for now as you said.

Reply:
De-coupled from main loop https://github.com/aquasecurity/tracee/pull/725/commits/93f7ce2aae884a80a52aad0590387e9042770676

Reply:
see https://github.com/aquasecurity/tracee/pull/725/files/4cb2e9a8f852d01a77429611a5dd0e525e28d87f#r642904379